src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ObjectOrCollectionInitializerExpression.cs

```diff
@@ -663,7 +663,9 @@ private BoundExpression EvaluateSideEffects(BoundExpression arg, RefKind refKind
 
 #if DEBUG
             var discardedUseSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
-            Debug.Assert(_compilation.Conversions.ClassifyConversionFromType(rewrittenReceiver.Type, memberSymbol.ContainingType, isChecked: false, ref discardedUseSiteInfo).IsImplicit);
+            // PROTOTYPE(RefStructInterfaces): Test various flavors of object/collection initializers on a type parameter that allows ref struct
+            Debug.Assert(_compilation.Conversions.ClassifyConversionFromType(rewrittenReceiver.Type, memberSymbol.ContainingType, isChecked: false, ref discardedUseSiteInfo).IsImplicit ||
```

Comment:
>_compilation.Conversions.ClassifyConversionFromType(rewrittenReceiver.Type, memberSymbol.ContainingType, isChecked: false, ref discardedUseSiteInfo).IsImplicit || [](http://example.com/codeflow?start=25&length=162)

It looks like this check can be dropped since it is included in `HasImplicitConversionToOrImplementsVarianceCompatibleInterface()`.

Reply:
> It looks like this check can be dropped since it is included in ```HasImplicitConversionToOrImplementsVarianceCompatibleInterface()```.

It actually isn't. ```HasImplicitConversionToOrImplementsVarianceCompatibleInterface``` deals only with interfaces.